action.yml

```diff
@@ -0,0 +1,243 @@
+      shell: bash
+      if: steps.extract-tag-exists.outputs.value == 'false' && steps.extract-version.outputs.value != 'false'
+      run: |
+        # Get the release tag for comparison
+        PREVIOUS_TAG=$(git describe --tags $(git rev-list --tags --max-count=1) 2>/dev/null || echo "")
+
+        # If there is no previous tag, then we can't compare
+        if [ -z "$PREVIOUS_TAG" ]; then
+          echo "value=false" >> $GITHUB_OUTPUT
+        else
+          echo "value=$PREVIOUS_TAG" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Check if the project has front-end assets
+      shell: bash
+      id: extract-has-node-assets
+      run: |
+        [[ -f package.json ]] && echo "value=true" >> $GITHUB_OUTPUT || echo "value=false" >> $GITHUB_OUTPUT
+
+    - name: Set environment variables
+      shell: bash
+      run: |
+        echo "GITHUB_TOKEN=$([ ! -z "${{ inputs.github-token }}"] && echo "${{ inputs.github-token }}" || echo "${{ github.token }}")" >> $GITHUB_ENV
+        echo "SKIP_BUILD=$([ "${{ inputs.skip-build }}" = "true" ] && echo "true" || echo "false")" >> $GITHUB_ENV
+        echo "SKIP_COMPOSER_INSTALL=$([ "${{ inputs.skip-composer-install }}" = "true" ] && echo "true" || echo "false")" >> $GITHUB_ENV
+        echo "SKIP_NPM_INSTALL=$([ "${{ inputs.skip-npm-install }}" = "true" ] && echo "true" || echo "false")" >> $GITHUB_ENV
+        echo "BUILT_BRANCH=${{ steps.extract-branch.outputs.value }}-built" >> $GITHUB_ENV
+        echo "VERSION_TAG=$([ ! -z "${{ steps.extract-tag-name.outputs.value }}" ] && echo "${{ steps.extract-tag-name.outputs.value }}" || echo "false")" >> $GITHUB_ENV
+    - name: Composer Install
+      if: inputs.skip-composer-install != 'true'
+      uses: alleyinteractive/action-test-php@develop
+      with:
+        php-version: '${{ inputs.php-version }}'
+        skip-audit: 'true'
+        skip-services: 'true'
+        skip-test: 'true'
+        skip-wordpress-install: 'true'
+
+    - name: NPM Install and Build
+      uses: alleyinteractive/action-test-node@develop
+      if: steps.extract-has-node-assets.outputs.value == 'true'
+      with:
+        build-command: '${{ inputs.build-command }}'
+        node-version: '${{ inputs.node-version }}'
+        skip-audit: 'true'
+        skip-test: 'true'
+
+    - name: Push to ${{ steps.extract-branch.outputs.value }}-built branch
+      if: steps.extract-is-built-branch.outputs.value == 'false' && steps.extract-branch.outputs.value != 'false' && github.repository != 'alleyinteractive/create-wordpress-plugin'
+      shell: bash
+      env:
+        CURRENT_BRANCH: '${{ steps.extract-branch.outputs.value }}'
+        CREATING_RELEASE: ${{ steps.extract-version.outputs.value != 'false' && steps.extract-tag-exists.outputs.value == 'false' && 'true' }}
+      run: |
+        git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
+        git config --global user.name "$GITHUB_ACTOR"
+
+        rm -rf .gitignore docker_tag output.log
```

Comment:
should we leave .gitignore if there's no `.deployignore` and no `.distignore`? 🤔

Reply:
true that